### ISSUE TITLE
What do I need to do for Math equations?

### ISSUE THREAD

**Issue by bjthomson** [lunet-io/markdig#259]
Firstly, nice parser. Thank you. I'm struggling to get the Mathematics extension working, however.

I dont know Latex (I'm a markdown novice too), but I tried the test example,  

    $$
    \begin{equation}
      \int_0^\infty \frac{x^3}{e^x-1}\,dx = \frac{\pi^4}{15}
      \label{eq:sample}
    \end{equation}
    $$
and it didn't show the integral symbol
I also tried the StackEdit samples:

    The *Gamma function* satisfying $\Gamma(n) = (n-1)!\quad\forall n\in\mathbb N$ is via the Euler integral

    $$
    \Gamma(z) = \int_0^\infty t^{z-1}e^{-t}dt\,.
    $$

StackEdit shows this:  
![mathquery](https://user-images.githubusercontent.com/22427749/47488354-1793ce00-d83c-11e8-89a3-974fa678bb66.PNG)
I saw that Markdown Monster used MarkDig, but that doesn't render the equation either.   
![mathquery3](https://user-images.githubusercontent.com/22427749/47489904-8a527880-d83f-11e8-8d0d-6c50a97b0a2f.PNG)

StackEdit mentions KaTeX, but I cant see a reference to that in the source. Looking at the page source in StackEdit, there is a bunch of HTML markup referring to KaTeX, but in my small test page, all I see is a div of class math?

![mathquery2](https://user-images.githubusercontent.com/22427749/47489359-5dea2c80-d83e-11e8-9782-b44ec4464d13.PNG)

I guess I'm missing something.
TIA

**Comment by xoofx** [comment]
So as you checked, markdig only generate a proper class/tag to detect math blocks. but it won't provide the rendering infrastructure. You need to plug an existing javascript Math framework for this (like KaTeX) and plug it into your final rendering page. Markdig doesn't generate a HTML page, it just generates a fragment of it.

Using with Katex, it should be something like this to process it (haven't tried this, but it should not be far from a version working):

```
<script>
  window.onload = function() {
      var tex = document.getElementsByClassName("math");
      Array.prototype.forEach.call(tex, function(el) {
          katex.render(el.textContent, el);
      });
  };
</script>
```

**Comment by bjthomson** [comment]
Ah I see, Thanks, i'll investigate.

**Comment by jonkas2211** [comment]
Hallo @xoofx,

shouldn't that be documented in the spec file?
For example something like:
Note that you have to use a rendering engine like 
```
<script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML' async></script>
```

**Comment by xoofx** [comment]
> shouldn't that be documented in the spec file?

I would prefer this in a proper documentation page instead, but I have had any spare time to bring documentation setup for markdig. I will leave this issue open until then.

**Comment by xoofx** [comment]
Likely fixed by #311

**Comment by tingjiangcao** [comment]
> 因此，正如您所检查的，markdig 仅生成适当的类/标签来检测数学块。但它不会提供渲染基础结构。您需要为此插入现有的 javascript 数学框架（如 KaTeX）并将其插入到最终渲染页面中。Markdig 不会生成 HTML 页面，它只会生成其中的一个片段。
> 
> 与 Katex 一起使用，应该是像这样来处理它（还没有尝试过，但它应该离工作版本不远）：
> 
> ```
> <script>
>   window.onload = function() {
>       var tex = document.getElementsByClassName("math");
>       Array.prototype.forEach.call(tex, function(el) {
>           katex.render(el.textContent, el);
>       });
>   };
> </script>
> ```

If you want to display the rendering results in real time and keep the scrolling progress. Then with this method, it will cause the page to flicker.

You need to replace the ‘math’ tag directly in the ‘html’ of the markdig rendering result. This may need to be easily implemented in a web project (npm+KaTex).

I would suggest if it is possible to include common functions in these extensions directly in markdig, because many developers may use it to develop WPF and other applications, and they may not all be familiar with web development.

Thank you for your efforts.
